src/sdk/services/oauthService.ts

```diff
@@ -24,11 +22,28 @@ class OAuthService {
     return originUrl.origin.replace(regex[1], 'accounts');
   };
 
-  private getOAuthAuthorizeUrl = (api: ManifestApi) => {
+  private getOAuthAuthorizeUrl = (api: ManifestApi, redirectUri?: string) => {
     const host = this.getOAuthHost();
-    const scopes = encodeURIComponent(api.scopes.join(' '));
-    const redirectUri = encodeURIComponent(api.redirectUri);
-    return `${host}/oauth/authorize?client_id=${api.applicationId}&redirect_uri=${redirectUri}&response_type=code&scope=${scopes}`;
+    const scopes = encodeURIComponent(api.scopes.join(','));
+    const selectedRedirectUri = encodeURIComponent(this.selectRedirectUri(api, redirectUri));
+    const clientId = api.client.id ?? api.applicationId;
+    return `${host}/oauth/authorize?client_id=${clientId}&redirect_uri=${selectedRedirectUri}&response_type=code&scope=${scopes}`;
+  };
+
+  private selectRedirectUri = (api: ManifestApi, redirectUri?: string) => {
+    if (redirectUri === undefined) {
+      if (api.redirectUris.length) {
```

Comment:
What if `api.redirectUris` is `undefined` ?

Reply:
We have TypeScript and runtime validation to help us ensure it's not. But I can add an additional check.

Reply:
Added

Reply:
This is related to [the comment](https://github.com/getoutreach/extensibility-sdk/pull/31#discussion_r878063506) that would make all deprecated props optional